cross/libgpg-error/Makefile

```diff
@@ -11,6 +11,13 @@ HOMEPAGE = http://www.gnupg.org/related_software/libgpg-error/
 COMMENT  = Libgpg-error is a small library that defines common error values for all GnuPG components
 LICENSE  = LGPL
 
+PRE_CONFIGURE_TARGET = preConfigureTarget
 GNU_CONFIGURE = 1
 
 include ../../mk/spksrc.cross-cc.mk
+
+.PHONY: preConfigureTarget
+preConfigureTarget:
+	cp $(WORK_DIR)/$(PKG_DIR)/src/syscfg/lock-obj-pub.arm-unknown-linux-gnueabi.h $(WORK_DIR)/$(PKG_DIR)/src/syscfg/lock-obj-pub.linux-gnueabi.h
```

Comment:
If you get an actual error during cross-compiling which tells you what name is required, then it seems easy to me: Why not just compile all arches (`make all-supported && `make all-legacy`), and jot down the file names that are generated, and the file names that are required?

If the names are all built up the same way, there's got to be a way to catch that in a couple of lines of code, while being as future proof as possible.

Your linked code, as well as the original code it was based off (http://cgit.openembedded.org/openembedded-core/tree/meta/recipes-support/libgpg-error/libgpg-error_1.19.bb#n23) can probably serve as inspiration.